File: packages/go/organizations.go
package hubfly

import (
	"context"
	"fmt"
)

type OrganizationsService struct {
	client *Client
}

func (s *OrganizationsService) List(ctx context.Context) (*Response[[]Organization], error) {
	return Request[[]Organization](ctx, s.client, "GET", "/api/v1/organizations", nil)
}

func (s *OrganizationsService) Get(ctx context.Context, orgID string) (*Response[Organization], error) {
	path := fmt.Sprintf("/api/v1/organizations/%s", orgID)
	return Request[Organization](ctx, s.client, "GET", path, nil)
}

func (s *OrganizationsService) Create(ctx context.Context, name string) (*Response[Organization], error) {
	return Request[Organization](ctx, s.client, "POST", "/api/v1/organizations", map[string]string{
		"name": name,
	})
}

func (s *OrganizationsService) ListMembers(ctx context.Context, orgID string) (*Response[[]OrgMember], error) {
	path := fmt.Sprintf("/api/v1/organizations/%s/members", orgID)
	return Request[[]OrgMember](ctx, s.client, "GET", path, nil)
}

File: packages/ts/src/modules/organizations.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse, Organization, OrgMember } from '../types.js';

export class OrganizationsModule {
  constructor(private readonly client: HubflyClient) {}

  /**
   * List user's organizations
   */
  public async list(): Promise<ApiResponse<Organization[]>> {
    return this.client.request<Organization[]>('/api/v1/organizations');
  }

  /**
   * Get organization by ID
   */
  public async get(orgId: string): Promise<ApiResponse<Organization>> {
    return this.client.request<Organization>(`/api/v1/organizations/${orgId}`);
  }

  /**
   * Create a new organization
   */
  public async create(name: string): Promise<ApiResponse<Organization>> {
    return this.client.request<Organization>('/api/v1/organizations', {
      method: 'POST',
      body: { name },
    });
  }

  /**
   * List members of an organization
   */
  public async listMembers(orgId: string): Promise<ApiResponse<OrgMember[]>> {
    return this.client.request<OrgMember[]>(`/api/v1/organizations/${orgId}/members`);
  }
}
